pkg/apis/pipeline/v1beta1/taskrun_validation.go

```diff
@@ -93,6 +95,43 @@ func (ts *TaskRunSpec) Validate(ctx context.Context) (errs *apis.FieldError) {
 	return errs
 }
 
+// validateInlineParameters validates parameters that are defined inline.
+// This is cruicial for propagated parameters since the parameters could
+// be defined under taskRun and then called directly in the task steps.
+// In this case, paratemers cannot be validated by the underlying taskSpec
+// since they may not have the parameters declared because of propagation.
+func (ts *TaskRunSpec) validateInlineParameters(ctx context.Context) (errs *apis.FieldError) {
```

Comment:
Suggested first line of docstring: "validateInlineParameters validates that any parameters called in the Task spec are declared in the TaskRun. this is crucial for propagated parameters because...." (This makes it clearer what is valid vs invalid)